app/products/calls/client/rest.ts

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {ServerChannelState, ServerConfig} from '@calls/types/calls';
+
+export interface ClientCallsMix {
+    getEnabled: () => Promise<Boolean>;
+    getCalls: () => Promise<ServerChannelState[]>;
+    getCallsConfig: () => Promise<ServerConfig>;
+    enableChannelCalls: (channelId: string) => Promise<ServerChannelState>;
+    disableChannelCalls: (channelId: string) => Promise<ServerChannelState>;
+}
+
+const ClientCalls = (superclass: any) => class extends superclass {
+    getEnabled = async () => {
+        try {
+            await this.doFetch(
+                `${this.getCallsRoute()}/version`,
+                {method: 'get'},
+            );
+            return true;
+        } catch (e) {
+            return false;
```

Comment:
I think the caller of `getEnabled` only cares if the calls plugin is reachable at the moment. (The ChannelInfo options/buttons would then get the `(Not available)` postfix. If they press it again, it will try again, and maybe remove the postfix if the call succeeded). We can add an error alert if needed?

Reply:
not needed, I get how it works now... thanks for explaining